jetbrains-core/src/software/aws/toolkits/jetbrains/services/ecs/ContainerActions.kt

```diff
@@ -124,10 +135,17 @@ class ContainerLogsAction(
 class ExecuteCommandAction(
     private val project: Project,
     private val container: ContainerDetails
-) : AnAction(message("ecs.execute_command_run"), null, null) {
+) : AnAction(message("ecs.execute_command_run"), null, null),
+    CoroutineScope by ApplicationThreadPoolScope("ContainerActions") {
     override fun actionPerformed(e: AnActionEvent) {
-        SessionManagerPluginInstallationVerification.requiresSessionManager(project) {
-            RunCommandDialog(project, container).show()
+        launch {
+            EcsExecUtils.ensureServiceIsInStableState(project, container.service, message("ecs.execute_command_run")) {
+                SessionManagerPluginInstallationVerification.requiresSessionManager(project) {
+                    runInEdt {
```

Comment:
Since we are already in a coroutine, we should just use withContext here with an EDT context:
`software.aws.toolkits.jetbrains.utils.CoroutineUtilsKt#getCoroutineUiContext`

Reply:
not made this change, since withContext would require the function to be a suspend function